src/plugins/manual_control/manual_control_impl.cpp

```diff
@@ -28,6 +28,13 @@ void ManualControlImpl::disable() {}
 
 void ManualControlImpl::start_position_control_async(const ManualControl::ResultCallback callback)
 {
+    if (_input == Input::NotSet) {
+        if (callback) {
+            callback(ManualControl::Result::Unknown);
+        }
+        return;
+    }
```

Comment:
Hmm I think it should be called more [like this](https://github.com/mavlink/MAVSDK/blob/ba2d819a3f6b2c0ea0279f6b45082e844c547051/src/plugins/manual_control/manual_control_impl.cpp#L144-L145) (I haven't tested it, though):

```suggestion
    if (_input == Input::NotSet) {
        if (callback) {
            auto temp_callback = callback;
            _parent->call_user_callback(
                [temp_callback]() { temp_callback(ManualControl::Result::Unknown); });
        }
        return;
    }
```

@julianoes: does that seem correct?

Reply:
Should I make the changes or wait for Julian's review?

Reply:
You can make the change (and test it, I did not :see_no_evil:  :innocent:).

Reply:
I tested it. Both styles of callback are working. If the method that you suggest is preferred due to any reason I will make the change and commit

Reply:
Yes it is preferred, because then it goes through the event loop instead of keeping the thread for itself. Something like that :smile:.